src/fastapi_app/api_models.py

```diff
@@ -17,3 +17,15 @@ class ThoughtStep(BaseModel):
     title: str
     description: Any
     props: dict = {}
+
+
+class RAGContext(BaseModel):
+    data_points: dict[int, dict[str, Any]]
+    thoughts: list[ThoughtStep]
+    followup_questions: list[str] | None = None
+
+
+class RetrievalResponse(BaseModel):
+    message: Message
+    context: RAGContext
+    session_state: Any | None = None
```

Comment:
This is the same as the frontend types to avoid dicts and automatic validation from pydantic.